File: 0001_Two_Sum/0001_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i=0
        l=[]
        while(i<len(nums)):
            if target-nums[i] in nums:
                l.append(i)
                l.append(nums.index(target-nums[i]))
            i+=1
        l=list(set(l))
        return l

File: 0001_Two_Sum/0001_Two_Sum.cpp
class TwoSum {
    private:
        unordered_map<int, int> nums;
    public:

        //O(1) add
        void add(int number) {
            nums[number]++;
        }

        //O(n) find
        bool find(int value) {
            int one, two;
            for(auto it = nums.begin(); it != nums.end(); it++){
                one = it->first;
                two = value - one;
                if ( (one == two && it->second > 1) ||
                        (one != two && nums.find(two) != nums.end() ) ){
                    return true;
                }
            }
            return false;
        }
};